Topics/Software_Tools/fish.md

```diff
@@ -0,0 +1,80 @@
+# A fin-troduction to `fish`
+
+## Introduction
+
+### enter `fish`!
+
+...the **f**riendly **i**nteractive **sh**ell.
+
+fish offers a command-line interface that is focused on usability and interactive use[^1].
+
+This introduction will be less about the syntax differences between the fish language
+and bash or zsh and more about the out-of-the-box features that are offered from this shell.
+
+### A dish served best straight out of the oven
+
+Some of you might be familiar with zsh plugin managers such as
+[oh-my-zsh](https://ohmyz.sh/) that can configure zsh to your liking, but fish
+just works. No more messing with chunky zsh configs and slow shells that make
+you wait several hundred milliseconds to type another command.
+
+### Installation
+
+It's recommended to check the [fish homepage](https://fishshell.com/) for up-to-date
+instructions for installing the latest version of fish on your platform.
+
+## Syntax highlighting
+
+fish performs syntax highlighting as you type. Invalid or non-existent, unexecutable
+commands appear red if they cannot be executed. On the contrary, valid commands will
+appear in a different color.
+
+## Autosuggestions
+
+fish suggests commands as you type, based on command history. As you type commands,
+suggestions are offered after the cursor in gray. To accept the suggestion, press
+`→` or `Control`+`F`.
+
+<img width="280" alt="autosuggestions" src="https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/40929644/fd505f17-4265-41ac-b41f-59bbdd0eb670">
+
```

Comment:
You could present an example here about the example autocompletions that come with fish.

Reply:
Sure, I can include a screenshot in "Autocompletions" as well.